ZCRMiOS/Classes/ZCRMDashBoardComponent.swift

```diff
@@ -0,0 +1,681 @@
+//
+//  ZCRMDashBoardComponent.swift
+//  Pods
+//
+//  Created by Kalyani shiva on 22/08/18.
+//
+
+
+import Foundation
+
+public class ZCRMDashBoardComponent {
+    
+    fileprivate var name = String()
+    fileprivate var category = ComponentCategory.none
+    fileprivate var reportId:Int64?
+    fileprivate var componentMarker: [ComponentMarkers]?
+    fileprivate var lastFetchedTimeLabel: String?
+    fileprivate var lastFetchedTimeValue: String?
+    fileprivate var componentChunks = [ComponentChunks]()
+    
+    
+    //MARK: Component Visualisation Props
+    
+    fileprivate var maximumRows: Int?
+    fileprivate var objective: Objective?
+    fileprivate var componentType = String()
+    fileprivate var colorPaletteName: colorPalette?
+    fileprivate var colorPaletteStartingIndex: Int?
+    fileprivate var segmentRanges: [SegmentRanges]?
+    
+}
+
+//MARK:- Enums
+
+extension ZCRMDashBoardComponent {
+    
+    public typealias colorPalette = ZCRMDashBoardComponentColorThemes.ColorPalette
+    
+    public enum Objective: String {
+        
+        case increase = "increase"
+        case decrease = "decrease"
+        
+    }
+    
+    public enum ComponentCategory: String {
+        
+        case chart = "chart"
+        case kpi = "kpi"
+        case comparator = "comparator"
+        case anomalyDetector = "trends"
+        case targetMeter = "target_meter"
+        case funnel = "funnel"
+        case none = "default Value"
```

Comment:
none is needed?

Reply:
Yes Anna . Firstly I did not set it to an OPTIONAL TYPE because we always receive a value for Component Type . Second , If its a non optional type , I wanted to give it a default value and not any component type . So, I introduced NONE type

Reply:
we have to give what we received don't introduce a new type or default type in our SDK.